mydiff.diff

```diff
@@ -0,0 +1,249 @@
++
++// |Push-pull
++// |User revoked access to channel
++// +
++// Sync Function includes 'requireAccess(revokedChannel)'
++// |Previously synced documents are auto purged on Couchbase Lite.
++// +
++// Local changes continue to be  pushed to remote but are rejected by Sync Gateway
++
++// |===
++
++// Clients may specify pull-filters to override the auto-purge behavior -- see: <<tbl-pull-filters>>.
++// This ensures backwards compatible with 2.8 clients that use pull filters to prevent auto purge of removed docs.
++
++// .Impact of Pull-Filters
++// [#tbl-pull-filters,cols="^1,2,2"]
++// |===
++
++// .2+.^h|purge_on_removal setting
++
++// 2+^h|Pull Filter
++
++// ^h|Not Defined
++// ^h|Defined to filter removals/revoked docs
++
++// |disabled
++// 2+a|Doc remains in local database
++
++// App notified of “accessRemoved” if a _Documentlistener_ is registered
++
++// |enabled (DEFAULT)
++// a|Doc is auto purged
++
++// App notified of “accessRemoved” if _Documentlistener_ registered
++// a|Doc remains in local database
++
++// App notified of “Rejected Rev(403)” if _Documentlistener_ registered
++
++// |===
++
++
++[#lbl-isgr]
++== Inter-Sync Gateway
++
++=== Access Revoked
++
++NOTE: This behavior is the *reverse* of that between {sgw-t} and Couchbase Lite -- see: <<lbl-sgw-cbl>>.
++
++// Document auto-purge behavior in inter-Sync Gateway replications following channel access revocation.
++
++By default, documents are *not* auto purged on the active sync gateway even if the user on the passive sync gateway loses channel access.
++
++You can opt-in to auto-purge behavior using the replicator level option `purge_on_removal` in the REST API -- see {rest-api-admin-isgr--xref-purge-on-removal}.
++
++Documents will then *be* auto-purged -- on active Sync Gateway nodes that have opted-in -- if they do not belong to *any* of the replicating user’s footnote:[pass:q,a[The _replicating user_ is the user on the _passive_ sync gateway cluster; the user specified in the replication definition.]] channels -- see: <<lbl-isgr-revoke-behaviors>>.
++
++
++// [#tbl-isgr-revoke-behaviors,cols="^1,^2,3,4"]
++// |===
++
++// .2+^.^h|Direction
++// 2+^h|System State
++// ^h|Impact on Sync
++
++// h|Active Sync Gateway (Local -- Running as admin user)
++// h|Passive Sync Gateway (Remote)
++// ^h|Expected behavior when `purge_on_removal=true`
++// .Access Regained behavior
++[#lbl-isgr-revoke-behaviors]
++.Access Revocation behavior
++====
++Access control policies are only enforced at the remote cluster.
++
++Here the Active Sync Gateway (Local) is running as an admin user with  `purge_on_removal=true`
++
++[#tbl-isgr-revoke-behaviors,cols="^1a,3a,6a", options="header"]
++|===
++
++// .2+^.^h
++|Direction
++|Passive Sync Gateway (Remote)
++|Expected Sync behavior
++
++|Pull
++|User revoked access to channel
++|Previously synced documents are auto purged on local
++
++|Push
++|User revoked access to channel
++|Previously synced documents are auto purged on local
```

Comment:
Push replications do not result in any revocations or purges occurring.

Reply:
Fixed